packages/helpwave_service/lib/src/auth/authentication_service.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:url_launcher/url_launcher.dart';
+import 'package:openid_client/openid_client_io.dart';
+
+class AuthenticationService {
+  static final _authenticationService = AuthenticationService._internal();
+
+  AuthenticationService._internal();
+
+  factory AuthenticationService() => _authenticationService;
+
+  urlLauncher(Uri url) async {
+    await launchUrl(url, mode: LaunchMode.inAppWebView);
+  }
+
+  authenticate(
+    String discoveryUrl,
+    String redirectUrl, {
```

Comment:
The redirectUrl should be non-configurable by the caller due to the fact that the caller should not create its own callback server.